src/components/search/Filter/Filter.tsx

```diff
@@ -32,154 +30,34 @@ interface FilterProps {
-  useEffect(() => {
-    if (isOpen) {
-      return
-    }
-
-    setActiveFacets([])
-    setSelectedFacets([...searchState.selectedFacets])
-  }, [isOpen, searchState.selectedFacets])
-
-  // Opens accordion items with active facets
-  useEffect(() => {
-    // Ensures all the active facets were identified
-    if (activeFacets.length !== filteredFacets.length) {
-      return
-    }
-
-    // Ensures there isn't empty facets
-    const selectedActiveFacets = activeFacets.filter(
-      (item) => item.facets.length > 0
-    )
-
-    // Checks if accordion item is already opened
-    selectedActiveFacets.forEach(
-      ({ accordionIndex }) =>
-        !indicesExpanded.has(accordionIndex) &&
-        onAccordionChange(accordionIndex)
-    )
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [isOpen, activeFacets])
-
-  const onFacetChange = (item: IStoreSelectedFacet) => {
-    const indexToRemove = selectedFacets.findIndex(
-      (facet) => facet.value === item.value
-    )
-
-    const shouldRemoveFacet = indexToRemove !== -1
-
-    setSelectedFacets((previousSelectedFacets) => {
-      if (shouldRemoveFacet) {
-        previousSelectedFacets.splice(indexToRemove, 1)
-
-        return [...previousSelectedFacets]
-      }
-
-      return [...previousSelectedFacets, item]
-    })
-  }
-
-  const onAccordionItemMount = (
-    index: number,
-    values: Filter_FacetsFragment['values']
-  ) => {
-    // Ensures only one array item for each accordion's item
-    if (activeFacets.length >= filteredFacets.length) {
-      return
-    }
-
-    // Filter current selected facets from API
-    const selectedValues = values.filter(({ selected }) => selected)
-
-    activeFacets.push({
-      accordionIndex: index,
-      facets: selectedValues.map(({ value }) => value),
-    })
-    setActiveFacets(activeFacets)
-  }
-
-  const onApply = () => {
-    // Only toggle new facets added and old facets removed and keep the current ones applied
-    const facetsToAdd = filterFacetsDiff(
-      selectedFacets,
-      searchState.selectedFacets
-    )
-
-    const facetsToRemove = filterFacetsDiff(
-      searchState.selectedFacets,
-      selectedFacets
-    )
-
-    toggleFacets([...facetsToAdd, ...facetsToRemove])
-    dismissTransition.current?.()
-  }
+  const { facets, selected, expanded, dispatch } = useFilter(allFacets)
 
   return (
     <>
       <div className="hidden-mobile">
         <Facets
+          facets={facets}
           testId={`desktop-${testId}`}
-          selectedFacets={selectedFacets}
-          filteredFacets={filteredFacets}
-          indicesExpanded={indicesExpanded}
+          indicesExpanded={expanded}
           onFacetChange={toggleFacet}
-          onAccordionChange={onAccordionChange}
-          onAccordionItemMount={onAccordionItemMount}
+          onAccordionChange={(index) =>
+            dispatch({ type: 'toggleExpanded', payload: index })
+          }
```

Comment:
Make sense to wrap it as a callback function to prevent re-render the Facets component?

Reply:
I don't think it makes any difference on this case, but I can do it

Reply:
Yeah, I tried doing this now but had some issues, let's leave it like this